static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx

```diff
@@ -242,72 +352,234 @@ export function useVirtualizedTree<T extends TreeLike>(
     return () => {
       scrollContainer.removeEventListener('scroll', handleScroll);
     };
-  }, [props.scrollContainer]);
+  }, [props.scrollContainer, props.rowHeight]);
+
+  useEffect(() => {
+    const scrollContainer = props.scrollContainer;
+
+    if (!scrollContainer) {
+      return undefined;
+    }
+
+    // Because nodes dont span the full width, it's possible for users to
+    // click on a node at the far right end which is outside of the row width.
+    // In that case, check if the top position where the user clicked overlaps
+    // with a row and select that row.
+    const handleClick = (evt: MouseEvent) => {
+      if (evt.target !== scrollContainer) {
+        // user clicked on an element inside the container, defer to onClick
+        return;
+      }
+
+      const rect = (evt.target as HTMLDivElement).getBoundingClientRect();
+      const index = Math.floor(
+        (latestStateRef.current.scrollTop + evt.clientY - rect.top) / props.rowHeight
+      );
+
+      // If a node exists at the index, select it
+      if (tree.flattened[index]) {
+        dispatch({type: 'set tab index key', payload: index});
+        updateGhostRow({
+          ref: clickedGhostRowRef,
+          tabIndexKey: index,
+          scrollTop: latestStateRef.current.scrollTop,
+          rowHeight: props.rowHeight,
+          backgroundColor: theme.blue300,
+        });
+      }
+    };
+
+    // Because nodes dont span the full width, it's possible for users to
+    // click on a node at the far right end which is outside of the row width.
+    // In that case, check if the top position where the user clicked overlaps
+    // with a row and select that row.
```

Comment:
This comment should talk about hovers and not clicks.

Reply:
it's actually both, I'll update it